src/lightning_lite/accelerators/accelerator.py

```diff
@@ -0,0 +1,73 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from abc import ABC, abstractmethod
+from typing import Any, Dict
+
+import torch
+
+
+class Accelerator(ABC):
+    """The Accelerator base class. An Accelerator is meant to deal with one type of hardware.
+
+    Currently, there are accelerators for:
+
+    - CPU
+    - CUDA (NVIDIA)
+    - MPS (Apple Silicon)
+    - TPU (Google)
+    """
+
+    @abstractmethod
+    def init_device(self, device: torch.device) -> None:
+        """Create and prepare the device for the current process."""
```

Comment:
Previously, this was named setup_environment. Renamed to `init_device` to make the intent of this method clear and focused. 

The rest of the interface was adapted 1:1 from PL. All methods are now marked abstract.

Reply:
> Does that mean you will rename setup_environment for the PL accelerators?

Yes. We need to make the hooks and APIs in Lite very focused and unambiguous. It is guided by the flexibility we need to maintain in Lite, as there are less assumptions than in PL. This also makes the code more readable. setup_environment was named because of where and how it was called in Lite, prior to accelerator being under strategy.

Reply:
Changed to setup_device